BaaS.md

```diff
@@ -0,0 +1,67 @@
+# Backup as a Service
+
+The Backup process entails a collection of ansible scripts that manipulate the servers to leave them in a “blank” state suitable for performing a consistent backup, a specific tool that is able to backup a complete Linux machine into an ISO image that later can be used to restore and an openstack CLI integration that allows the operator to perform the actions in a very easy way.
+
+This backup automation has a lot of things built in and it is very helpful, but it still lacks some degree of integration. To solve that, we are going to implement what we call backup as a service. This service will be a piece of software that will run on a container in the openstack’s overcloud, exposing an API that can be accessed by third parties to trigger or program backups, to list all the history of backups that are stored in the system and allow to trigger restorations or delete old backups. It will even be capable of accepting backup programming and programmed cleanups.
+
+As Openstack on Openshift deployed via an operator seems to be the future of Openstack, my proposal is to write this Service in Golang, which is much better aligned with the Openshift/Kubernetes/containers environment than Openstack’s traditional Python.
+
+## BACKUP
+
+The basic API that will be implemented in the service is a kind of a CRD (Create, Read, Delete) to manage the backups. The API will be restful and versioned, so five basic call URLs will look like this:
```

Comment:
Should we also include how the calls would be for the CLI.
For example :
`openstack overcloud backup list`

Reply:
This is a good idea, but I think this should be done on a "CLI spec". Please, add one such spec to this repo and create a PR to merge it. As the idea is to keep doing all the actions via ansible, both the CLI and the API should evolve at the same time with little extra effort.